torch/_inductor/ir.py

```diff
@@ -7412,65 +7430,93 @@ def create(  # type: ignore[no-untyped-def]
             len(all_inputs) > 0
         ), "torch.while_loop is assumed to have at least one operand."
 
-        device = all_inputs[0].get_device()
-
+        device = next(
```

Comment:
Below changes are pretty mechanical, since we allow symint/int carries, we need to make the code that previously only works for tensors to also work for symints/ints.